libexec/rtld-elf/mips/reloc.c

```diff
@@ -850,7 +850,7 @@ reloc_non_plt(Obj_Entry *obj, Obj_Entry *obj_rtld, int flags,
 			break;
 		case R_TYPE(CHERI_CAPABILITY):
 			if (process_r_cheri_capability(obj, r_symndx, lockstate,
```

Comment:
I would keep the comment that the first 8bytes are the addend for BE MIPS and not the second 8.

Reply:
The comment was about REL vs RELA, and so given we don't have any comments in this bit of code documenting that fact (given it's pretty obvious from the use of Elf_Rel everywhere, no r_addend and lots of loads) I figured it was superfluous. The comment mentioned it was the first 8 bytes, but did not highlight the fact that this is conceptually wrong for a BE target, as it should be in the last (ie either second or fourth) 8 bytes. I will add a new comment that makes *that* point.

Reply:
We could change that now that CHERI256 is no longer supported, but I think reducing churn for MIPS is a good idea.

Reply:
Or alternatively we should view this as a `union { size_t; uintcap_t; }`, and then the code becomes correct. Which is probably the conceptually-correct thing, even if we plumb through 128-bit values for use in some cases and introduce CBuildCap parts.